src/Sponsor/SponsorManagement.tsx

```diff
@@ -184,27 +188,23 @@ class ManageSponsorContainer extends React.Component<
         sponsorTier = -1;
     }
 
-    const sponsorApplication = this.convertFormikToSponsor(values, account.id, sponsorTier);
+    const sponsorApplication = this.convertFormikToSponsor(values, this.state.sponsorDetails.id, account.id, sponsorTier);
 
     switch (this.state.mode) {
       case ManageSponsorModes.CREATE:
         await Sponsor.create(sponsorApplication);
         break;
       case ManageSponsorModes.EDIT:
-        const sponsorResponse = await Sponsor.getSelf();
-        const sponsor = sponsorResponse.data.data;
-
-        sponsorApplication.id = sponsor.id;
         await Sponsor.update(sponsorApplication);
         break;
     }
   }
 
   private convertFormikToSponsor(
     values: FormikValues,
+    sponsorId: string = '',
```

Comment:
We should remove the default value if we're moving this so that it's before values with default